### ISSUE TITLE
Hide edit button and title on specified pages

### ISSUE THREAD

**Issue by Kanaduchi** [squidfunk/mkdocs-material#2163]
- [/] I've read the [contribution guidelines][1] and agree with them

__I want to suggest an idea and checked that ...__

- [/] ... to my best knowledge, my idea wouldn't break something for other users
- [/] ... the documentation does not mention anything about my idea
- [/] ... there are no open or closed issues that are related to my idea

## Description

In some cases it will be a great deal to hide page title and edit button on specific pages.  

Example:
```
---
hide:
  - pageTitle
  - editButton
--- 
```

**Comment by squidfunk** [comment]
Why? Could you elaborate on why one would want to do that? Also what do you mean by page title?

**Comment by Kanaduchi** [comment]
On some pages, e.g. on main pages I don't need any header and edit button. 

Currently to hide header I add empty H1. 
But is not good case because I still have big empty line at the top of the page:
![empty header](https://user-images.githubusercontent.com/7042994/103397750-77dcd380-4b5b-11eb-8f0b-f03241f5e4ff.PNG)

**Comment by squidfunk** [comment]
Providing an option for these kinds of tweaks is out-of-scope for this project – we just cannot make everything configurable. Hiding the sidebars involves some more complex logic, which is why these are provided as options. You can hide the headline and edit button with some inline CSS. Just add this to the Markdown file in question:

```
<style>
.md-typeset h1 {
  display: none;
}
.md-content__button {
  display: none;
}
</style>
```

**Comment by Kanaduchi** [comment]
@squidfunk inline css works like magic! Thank you!

**Comment by shenweiyan** [comment]
I found that using this inline CSS method also results in the hiding of the h1 titles in the search results of the current page.

**Comment by vinodgangal** [comment]
The inline css approach has a major pitfall, some how it removes/hides the version selection drop down also.

**Comment by alexvoss** [comment]
The solution to this is to make the CSS selector more specific. To make sure the page title is shown in the search results but not on the page:

```
.md-content .md-typeset h1 {
      display: none;
  }
```

I assume the same would go for the version selector but I don't use mike (I assume) myself. Give it  a try @vinodgangal and ping me if this does not work.

**Comment by vinodgangal** [comment]
@alexvoss I tried it, does't work. Still the same impact is seen that version drop down disappears.

**Comment by alexvoss** [comment]
Can you do a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) so I can gave a look? I am hoping to give versioning a try today anyway.

**Comment by ItsDrike** [comment]
Hey, so I've tried using the proposed workaround here of:

```css
.md-content .md-typeset h1 {
    display: none;
}
```

However, I'm running into an interesting issue. Basically, when I hide the title heading, mkdocs thinks that I'm "in that heading", which changes the header on the page. This wouldn't normally be as big of an issue, however, since I'm also using mike to version my docs, it hides the version selector too, which is annoying. See screenshots below:

<details>
<summary>With the custom CSS</summary>

<img width="1346" height="739" alt="Image" src="https://github.com/user-attachments/assets/daa58a1b-8a0c-4324-a2ae-bb902e4273d3" />

</details>

<details>
<summary>Without the custom CSS</summary>

<img width="1299" height="821" alt="Image" src="https://github.com/user-attachments/assets/7b680457-51ee-4e93-ab74-bc804dad3d95" />

</details>

---

After some debugging, I found that the `md-header__title--active` class is dynamically being inserted into the header div, which makes some CSS rules apply that take precedence and change which title div is visible. I found that this override is able to prevent this, however, I'm not that experienced with CSS, I don't know if this is a great way to do it.

```css
    /* Hide the h1 title */
    .md-content .md-typeset h1 { display: none; }

    /*
     * By hiding the h1 title, mkdocs will see it as active (scrolled at it)
     * which affects the header. This isn't what we want, as it makes the mike
     * version selector disappear, and it also just isn't what we'd want anyways.
     * Force the main header to be visible and hide the active one.
     */
    .md-header__title--active .md-header__topic {
        opacity: unset;
        pointer-events: unset;
        transform: unset;
        z-index: unset;
    }
    .md-header__title--active .md-header__topic + .md-header__topic {
        opacity: 0;
        pointer-events: unset;
        transform: unset;
        z-index: unset;
    }
```

If someone knows of a nicer way to handle this, or thinks that there's something wrong with this one, I'd love to hear some feedback on it.

That said, I think that at this point, it would be make more sense if mkdocs-material just had a switch to disable the title, just like you can disable navigation/toc, since this seems like a pretty annoying workaround to have, especially since it might be vulnerable to changes of the base theme CSS. I don't really understand the reason why this is considered out of scope, while hiding toc/navigation isn't.

**Comment by squidfunk** [comment]
@ItsDrike Fixed in 08fd86935. The version selector will now stay visible when applying the CSS to hide the page title.

**Comment by alexvoss** [comment]
> it would be make more sense if mkdocs-material just had a switch to disable the title

To add to what @squidfunk says: the title is rendered by Python Markdown and not by the Material for MkDocs templates. We'd need to inject ourselves into the Markdown rendering to change this or post-process the result. Neither option is very enticing, so I think that his fix is the right approach.

We are working on [foundational changes to Material](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/) that might well make such an option feasible, so that the CSS customization and this fix would no longer be necessary. 

In order to implement this as an option in the future we would need to understand precisely what the use case for this is. If anyone has a use case beyond "I don't want a title on the homepage" please let us know the reasoning.

**Comment by ItsDrike** [comment]
> In order to implement this as an option in the future we would need to understand precisely what the use case for this is. If anyone has a use case beyond "I don't want a title on the homepage" please let us know the reasoning.

For me, the reason is that I wanted to have a custom logo + text combination, which doesn't really work as an actual h1 title with mkdocs:

```markdown
<div style="display: flex; align-items: center; font-size: 75px; font-weight: bold; color: lightgray;">
    <img src="./assets/py-mine_logo.png" width=100 alt="Logo">
    MCPROTO
</div>
```

I could wrap this in `<h1>` but that doesn't register it as the page title, I could also move it into a single like and add `# ` before it, but that also doesn't seem to do the trick, and mkdocs adds in another h1 title before it. I assume this is because it contains HTML.

That said, for me, it is very much an aesthetic choice here, I don't like seeing Home title on the index page before this, since I want my own formatting there, and I do prefer no h1 whatsoever to the logo + text being the h1 too, as it looks better without the `#` symbol on the left, that's all. (I do think that most people would agree that the version without the Home h1 title looks better in the screenshots above.)

With the fix above, it's not as annoying to have to use custom CSS, I would still like a switch if it becomes more viable, since it is just more convenient, but I don't insist on it.

**Comment by squidfunk** [comment]
The fix was just released as part of [9.6.17](https://squidfunk.github.io/mkdocs-material/changelog/#9.6.17).
